### PR TITLE
Jakarta JSON Upgrades for Jakarta EE 10

### PR DESCRIPTION
**Pull request by jamezp**
* https://issues.redhat.com/browse/WFLY-16263
* https://issues.redhat.com/browse/WFLY-16264
* https://issues.redhat.com/browse/WFLY-16200
* https://issues.redhat.com/browse/WFLY-16262

Upgrade Jakarta JSON Processing to 2.1.0 and migrate to using Eclipse Parsson. Also upgrade Jakarta JSON Binding to 3.0.0 and Yasson to 3.0.0-RC1.

### CODE REVIEW

File: ee-9/common/src/main/resources/modules/system/layers/base/org/eclipse/parsson/main/module.xml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<!-- TODO Consider dropping direct dep on core-galleon-pack and instead using
+     a source dep, eliminating the need to re-provision this just to get a
+     different version. -->
```
Comment (bstansberry) [inline ee-9/common/src/main/resources/modules/system/layers/base/org/eclipse/parsson/main/module.xml:20]: I don't think this is relevant to this file. IIRC I wrote it somewhere. I doubt it's relevant there any more either. ;)

Comment (jamezp) [reply ee-9/common/src/main/resources/modules/system/layers/base/org/eclipse/parsson/main/module.xml:20]: Yeah. I almost removed it too. Then though what if it means something to core :) I can remove it though.

Comment (bstansberry) [reply ee-9/common/src/main/resources/modules/system/layers/base/org/eclipse/parsson/main/module.xml:20]: Well. I'm wrong. It's relevant to this file (assuming you'll add this module in core) but I guess isn't relevant to the org.glassfish.jakarta.json one.

If this module.xml file and the alias one are going into core with the same content, I believe both of these can be removed once the core upgrade with them in it is integrated. The TODO I was talking about is done.

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @jamezp

**Pull request merged**